common/functions/src/scalars/expressions/unary.rs

```diff
@@ -0,0 +1,43 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::marker::PhantomData;
+
+use common_datavalues2::prelude::*;
+use common_exception::Result;
+
+/// An common struct to caculate Unary expression scalar op.
+pub struct ScalarUnaryExpression<L: Scalar, O: Scalar, F> {
+    func: F,
+    _phantom: PhantomData<(L, O)>,
+}
+
+impl<'a, L: Scalar, O: Scalar, F> ScalarUnaryExpression<L, O, F>
+where F: Fn(L::RefType<'a>) -> O
```

Comment:
This bound looks not correct. F should be valid for all lifetime (e.g., `for<'a> F: Fn(L::RefType<'a>)`). If we implement expressions like this, we will need to create an expression every time we want to evaluate an array.

You may refer to day 6 "Erase Expression Lifetime" in https://github.com/skyzh/type-exercise-in-rust for solutions.

Reply:
Thank you @skyzh .
If so we should consider it in the later PR, cc @sundy-li

Reply:
Thanks, I get it, these will be adressed after we introduce datatypes into the creator function,  then we can have single expression to evaluate arrays multiple times.